src/drivers/intel/ssp/mn.c

```diff
@@ -188,14 +188,8 @@ static inline int set_mclk_divider(uint16_t mclk_id, uint32_t mdivr_val)
 	case 1:
 		mdivr = 0x00000fff; /* bypass divider for MCLK */
 		break;
-	case 2:
-		mdivr = 0x0; /* 1/2 */
-		break;
-	case 4:
-		mdivr = 0x2; /* 1/4 */
-		break;
-	case 8:
-		mdivr = 0x6; /* 1/8 */
+	case 2 ... 8:
+		mdivr = mdivr_val - 2; /* 1/n */
```

Comment:
@keyonjie I'm not convinced about this change. Can we please double check this is really ok ? It seems deliberate that we only allowed 2,4 and 8 previously

Reply:
@ranj063 yes, all 1/N (N=1, ..., 8, or even more) values are supported according to the spec. 

At the same time, since my PR has changed the mclk matching to start from the lowest frequency source, so in real case, bigger Ratio will be seldom used so the change here is only theoretical needed.